INSTALL.md

```diff
@@ -94,35 +89,61 @@ The benefits of this:
   - When you build your project, it will automatically update the headers if there is a change
   - Or get the specific version by setting it to a specific commit point
 
-
 ```cmake
-cmake_minimum_required(VERSION 3.0 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.8)
 project(FplusMinimalExternalExample)
-set(CMAKE_CXX_STANDARD 14)
 
 include(ExternalProject)
-ExternalProject_Add(functional_plus
-  GIT_REPOSITORY https://github.com/Dobiasd/FunctionalPlus.git
-  GIT_TAG master
+ExternalProject_Add(
+    functional_plus
+    GIT_REPOSITORY https://github.com/Dobiasd/FunctionalPlus.git
+    GIT_TAG master
 
-  SOURCE_DIR "${CMAKE_BINARY_DIR}/thirdparty/fplus"
+    SOURCE_DIR "${CMAKE_BINARY_DIR}/thirdparty/fplus"
 
-  CONFIGURE_COMMAND ""
-  BUILD_COMMAND ""
-  INSTALL_COMMAND ""
+    CONFIGURE_COMMAND ""
+    BUILD_COMMAND ""
+    INSTALL_COMMAND ""
 
-  LOG_DOWNLOAD ON
-  LOG_BUILD ON
+    LOG_DOWNLOAD ON
+    LOG_BUILD ON
 )
-set(FPLUS_INCLUDE_DIR ${CMAKE_BINARY_DIR}/thirdparty/fplus/include)
-include_directories(${FPLUS_INCLUDE_DIR})
+
+ExternalProject_Get_Property(functional_plus SOURCE_DIR)
 
 add_executable(main src/main.cpp)
 add_dependencies(main functional_plus)
+target_compile_features(main PRIVATE cxx_std_14)
+target_include_directories(main PRIVATE "${SOURCE_DIR}")
+```
+
+
+### way 3: using [CMake 3.14 FetchContent](https://cmake.org/cmake/help/latest/module/FetchContent.html)
+
+This should be preferred over the
+[ExternalProject](#way-2-using-cmakes-externalproject) method.
+
+This has similar benefits, but it's easier to set up in your CMakeLists.txt.
+
+```cmake
+cmake_minimum_required(VERSION 3.14)
+project(FplusMinimalExternalExample)
+
+include(FetchContent)
+FetchContent_Declare(
+    functional_plus
+    GIT_REPOSITORY https://github.com/Dobiasd/FunctionalPlus.git
+    GIT_TAG master
+)
+FetchContent_MakeAvailable(functional_plus)
+
+add_executable(main src/main.cpp)
+target_compile_features(main PRIVATE cxx_std_14)
```

Comment:
Isn't `cxx_std_14` set trhough the fplus target already?

Reply:
It's not wise to rely on interface requirements implicitly defining your own requirements.

Reply:
I see, the example uses C++14 internally, not to get fplus working.